packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts

```diff
@@ -108,6 +126,15 @@ export const useScaffoldEventHistory = <
     receiptData,
   ]);
 
+  useInterval(
+    async () => {
+      if (!deployedContractLoading) {
+        readEvents();
+      }
+    },
+    watch ? scaffoldConfig.pollingInterval : null,
```

Comment:
Maybe we make it `5000` if `targetNetwork`  is `hardhat` because it uses `30000` while on `hardhat` too, similar to what we have at : 

https://github.com/scaffold-eth/scaffold-eth-2/blob/44f31c17088d09f8428f0b2c6d51f8f5046b16a2/packages/nextjs/services/web3/wagmiConnectors.tsx#L39

Reply:
I'm not sure, because if you want faster polling when using hardhat you just change the value at the scaffoldConfig file. 

If you use some special value for hardhat and then the user starts using a real blockchain, and the events start loading slowly, maybe the user doesn't understand why they were loading faster before.

Reply:
> I'm not sure, because if you want faster polling when using hardhat you just change the value at the scaffoldConfig file.

The thing is that we don't use the scaffoldConfig polling value when we are on hardhat / foundry (check the wagmiConnectors file Shiv linked), since we want to be as fast as possible. The default seems to be 4000 ms https://wagmi.sh/react/providers/configuring-chains#pollinginterval-optional

> If you use some special value for hardhat and then the user starts using a real blockchain, and the events start loading slowly, maybe the user doesn't understand why they were loading faster before.

Yep, this might be true! But it's how we are doing it rn :/ We have a warning in scaffoldConfig tho:

```
  // The interval at which your front-end polls the RPC servers for new data
  // it has no effect on the local network
```

So for the sake of consistency, it might be a good idea to set 4000 for local chains?

What do you all think?

Reply:
Makes sense! Changed! Thanks!!